### PR TITLE
[schemes/Typography] Add useCurrentContentSizeCategoryWhenApplied.

### PR DESCRIPTION
**Pull request by jverkoey**
This new property is a replacement for `mdc_adjustsFontForContentSizeCategory`, with the name more clearly demonstrating its intended use case.

Expected client usage:

```objc
typographyScheme = [[MDCTypographyScheme alloc] initWithDefaults:
                        MDCTypographySchemeDefaultsMaterial201902];
typographyScheme.useCurrentContentSizeCategoryWhenApplied = YES;
containerScheme.typographyScheme = typographyScheme;
button.adjustsFontForContentSizeCategoryWhenScaledFontIsUnavailable = NO;
[button applyThemeWithScheme:containerScheme];
```

Expected theming extension implementation usage (pseudo-code):

```objc
- applyThemeWithScheme:(id<MDCContainerScheming>)scheme {
 if ([scheme.typographyScheme respondsToSelector:@selector(useCurrentContentSizeCategoryWhenApplied)]
         && scheme.typographyScheme.useCurrentContentSizeCategoryWhenApplied) {
    UIContentSizeCategory sizeCategory = UIContentSizeCategoryLarge;
    if (@available(iOS 10.0, *)) {
      sizeCategory = self.traitCollection.preferredContentSizeCategory;
    } else if ([UIApplication mdc_safeSharedApplication]) {
      sizeCategory = [UIApplication mdc_safeSharedApplication].preferredContentSizeCategory;
    }
    [self setTitleFont:[scheme.typographyScheme.button mdc_scaledFontForSizeCategory:sizeCategory]
              forState:UIControlStateNormal];
  } else {
    [self setTitleFont:scheme.typographyScheme.button forState:UIControlStateNormal];
  }
}
```

Issue is being mirrored. Will update this description once it is available.

### CODE REVIEW

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (yarneo) [inline components/schemes/Typography/src/MDCTypographyScheme.h]: I didn't fully understand this sentence, the "scheme to components" part is where I got stuck grammatically

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (yarneo) [inline components/schemes/Typography/src/MDCTypographyScheme.h]: tip: Add a "Note:" prefix

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (yarneo) [inline components/schemes/Typography/src/MDCTypographyScheme.h]: tip: Add a "Note:" prefix

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (yarneo) [inline components/schemes/Typography/src/MDCTypographyScheme.h]: same here, I didn't grammatically fully understand this sentence, specifically on the "to" word connecting both parts of the sentence.

File: components/schemes/Typography/tests/unit/MDCTypographySchemeTests.m
Comment (yarneo) [inline components/schemes/Typography/tests/unit/MDCTypographySchemeTests.m:152]: suggestion: add a test to make sure mdc_adjustsFontForContentSizeCategory is imitating the useCurrentContentSizeCategoryWhenApplied behavior.

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (jverkoey) [reply components/schemes/Typography/src/MDCTypographyScheme.h]: Took another stab at it. What do you think?

File: components/schemes/Typography/tests/unit/MDCTypographySchemeTests.m
Comment (jverkoey) [reply components/schemes/Typography/tests/unit/MDCTypographySchemeTests.m:152]: Good idea! Added tests.

**Review by yarneo — APPROVED** [review]

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (codeman7) [inline components/schemes/Typography/src/MDCTypographyScheme.h:108]: _Tip:_ _"replacement for @c mdc_adjustsFontForContentSizeCategory"_

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (codeman7) [inline components/schemes/Typography/src/MDCTypographyScheme.h:122]: _Tip:_
```diff
-  MDCTypographyScheming's useCurrentContentSizeCategoryWhenApplied is assigned to NO by default with this version.
+ @c useCurrentContentSizeCategoryWhenApplied defaults to @c NO.
```

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (codeman7) [inline components/schemes/Typography/src/MDCTypographyScheme.h:134]: _Tip:_
```diff
-  MDCTypographyScheming's useCurrentContentSizeCategoryWhenApplied is assigned to NO by default with this version.
+ @c useCurrentContentSizeCategoryWhenApplied defaults to @c YES.
```

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (codeman7) [inline components/schemes/Typography/src/MDCTypographyScheme.h]: _Tip:_ I believe this file uses explicit property annotations, maybe beneficially to add `assign`.

File: components/schemes/Typography/tests/unit/MDCTypographySchemeTests.m
Comment (codeman7) [inline components/schemes/Typography/tests/unit/MDCTypographySchemeTests.m:152]: Should we add documentation to these method?

**Review by codeman7 — APPROVED** [review]
Looks great! I was very nit-picky but overall LGTM! I don't know if the nit-picks are worth a follow up so I'll leave that up to you.

File: components/schemes/Typography/src/MDCTypographyScheme.h
Comment (jverkoey) [reply components/schemes/Typography/src/MDCTypographyScheme.h:122]: I think the latter is a bit less clear if someone doesn't know what a `useCurrentContentSizeCategoryWhenApplied` property is or where it's configured.

File: components/schemes/Typography/tests/unit/MDCTypographySchemeTests.m
Comment (jverkoey) [reply components/schemes/Typography/tests/unit/MDCTypographySchemeTests.m:152]: Generally-speaking unit tests shouldn't need docs if the method names are self-documenting. Do you feel these methods could use clarification?

**Comment by jverkoey** [comment]
Closes https://github.com/material-components/material-components-ios/issues/7467

**Pull request merged**
